dvc/remote/ssh/__init__.py

```diff
@@ -129,20 +133,41 @@ def ssh(self, path_info):
             password=self.password,
         )
 
-    def exists(self, path_infos):
-        single_path = False
+    def exists(self, path_info):
+        with self.ssh(path_info) as ssh:
+            return ssh.exists(path_info.path)
 
-        if not isinstance(path_infos, list):
-            single_path = True
-            path_infos = [path_infos]
+    def batch_exists(self, path_infos, callback):
+        def _exists(chunk_and_channel):
+            chunk, channel = chunk_and_channel
+            ret = []
+            for path in chunk:
+                try:
+                    channel.stat(path)
+                    ret.append(True)
+                except IOError as exc:
+                    if exc.errno != errno.ENOENT:
+                        raise
+                    ret.append(False)
+                callback.update(path)
+            return ret
 
         with self.ssh(path_infos[0]) as ssh:
-            results = [ssh.exists(path_info.path) for path_info in path_infos]
+            with ssh.mute_paramiko():
+                channels = ssh.open_max_sftp_channels()
+                max_workers = len(channels)
+
+                with ThreadPoolExecutor(max_workers=max_workers) as executor:
+                    paths = [path_info.path for path_info in path_infos]
```

Comment:
use a queue and sort by size? to do big files first.

Reply:
@shcheklein , how does this helps? is it to distribute the weight correctly between workers so each one has something significant to do?

Reply:
Sorting by size can't help in this case, since each operation here takes the same time, it's not upload/download. Queue can be useful anyway, just to make sure all the workers are busy. It should not be a major issue I hope.

Reply:
@shcheklein queue is obviously an approach superior to chunking. I think it could be done separately though. Let's stick to the original task here and create a new issue for chunks to queue refactor if you wish.

Reply:
@Suor agreed, not a big issue in this specific case anyway